core/src/test/scala/matryoshka/instances/fixedpoint/stream.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2014–2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package matryoshka.instances.fixedpoint
+
+import matryoshka.{ToIdOps => toIdOps, _}, Recursive.ops._
+
+import scala.Int
+
+import org.specs2.ScalaCheck
+import org.specs2.mutable._
+import scalaz._, Scalaz._
+
+class StreamSpec extends Specification with ScalaCheck {
+  // fixes implicit resolution
+  type IntStream[A] = (Int, A)
+
+  /** Continues the Fibonacci sequence from a seed pair of values.
+    */
+  val fib: Coalgebra[(Int, ?), (Int, Int)] = {
+    case (p2, p1) =>
+      val c = p1 + p2
+      (c, (p1, c))
+  }
+
+  /** Generates an infinite stream of the carrier value.
+    */
+  def constantly[A]: Coalgebra[(A, ?), A] = i => (i, i)
+
+  "fib" should {
+    "begin with 1" in {
+      // TODO: implement `take`
+      (0, 1).ana[Nu, IntStream](fib).project._1 must_== 1
+    }
+
+    "lazily generate the correct sequence" in {
+      // TODO: implement `take`
+      (0, 1).ana[Nu, IntStream](fib).project._2.project._2.project._2.project._2.project._2.project._1 must_== 13
```

Comment:
Hmm. You can't implement `take`, because `Nu[IntStream]` is always infinite, right?

This is a bit nicer, but only a bit:

```scala
  val fibƒ: Coalgebra[(Int, ?), (Int, Int)] = {
    case (a, b) =>
      val c = a + b
      (c, (b, c))
  }

  val fib = (0, 1).ana[Nu, IntStream](fibƒ)

  // NB: should be `head[T[_[_]], A]: T[(A, ?)] => A`, but types don't infer.
  def head[T[_[_]]: Recursive](xs: T[IntStream]): Int =
    xs.project._1

  def tail[T[_[_]]: Recursive](xs: T[IntStream]): T[IntStream] =
    xs.project._2

  // TODO: use a fold?
  def drop[T[_[_]]: Recursive](n: Int)(xs: T[IntStream]): T[IntStream] =
    if (n > 0) drop(n-1)(tail(xs)) else xs

  head(drop(5)(fib)) must_== 13
```